src/Illuminate/Database/Query/Builder.php

```diff
@@ -1427,31 +1441,34 @@ public function simplePaginate($perPage = 15, $columns = ['*'], $pageName = 'pag
      */
     public function getCountForPagination($columns = ['*'])
     {
-        $this->backupFieldsForCount();
-
-        $this->aggregate = ['function' => 'count', 'columns' => $columns];
+        if ($grouped = isset($this->groups)) {
+            $selectSql = $this->toSql();
+        }
 
-        $results = $this->get();
+        $this->backupFieldsForCount($grouped);
 
-        $this->aggregate = null;
+        if ($grouped) {
+            $this->from = $this->raw("({$selectSql}) as subquery");
+        }
 
-        $this->restoreFieldsForCount();
+        $total = $this->count($columns);
```

Comment:
I would think so, cause we can't return the value before we restore the backed up fields. Unless you have a way of restoring the fields and returning the value on a single line, I think this will be necessary